functests/performance/rt-kernel.go

```diff
@@ -59,4 +60,19 @@ var _ = Describe("[performance]RT Kernel", func() {
 
 	})
 
+	It("[test_id:28526][crit:high][vendor:cnf-qe@redhat.com][level:acceptance] Non worker-cnf node should not have RT kernel installed", func() {
+
+		By("Skipping test if cluster does not have another available worker node")
+		nonRTWorkerNode, err := nodes.GetNonCNFNode(testclient.Client, "worker")
+		Expect(err).ToNot(HaveOccurred())
+
+		if nonRTWorkerNode == nil {
```

Comment:
this check won't work (it will never be false): in `GetNonCNFNode` you init the nodes with `nodesWithResource := []corev1.Node{}`, which is an _empty_, but not _nil_ slice. So you need to check `len(nonRTWorkerNode) == 0` here.

Reply:
Yes, I have resolved this to check the len of slice.